Mage.Sets/src/mage/cards/k/KjeldoranGuard.java

```diff
@@ -0,0 +1,130 @@
+                new InvertCondition(new DefendingPlayerControlsCondition(snowLandFiler)),
+                new IsPhaseCondition(TurnPhase.COMBAT, false));
+
+        Ability ability = new ConditionalActivatedAbility(
+                Zone.BATTLEFIELD,
+                new KjeldoranGuardEffect(),
+                new TapSourceCost(),
+                snowLandAndCombatCondition);
+        ability.addTarget(new TargetCreaturePermanent());
+        this.addAbility(ability);
+    }
+
+    private KjeldoranGuard(final KjeldoranGuard card) { super(card); }
+
+    @Override
+    public Card copy() {
+        return new KjeldoranGuard(this);
+    }
+}
+
+class KjeldoranGuardEffect extends OneShotEffect {
+
+    KjeldoranGuardEffect() {
+        super(Outcome.Neutral);
+        staticText = "Target creature gets +1/+1 until end of turn." +
+                "When that creature leaves the battlefield this turn, sacrifice Kjeldoran Guard." +
+                "Activate only during combat and only if defending player controls no snow lands.";
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        if (game.getPermanent(source.getFirstTarget()) == null) { return false; }
+
+        // Target creature gets +2/+2 until end of turn.
+        BoostTargetEffect buffEffect = new BoostTargetEffect(1, 1, Duration.EndOfTurn);
+        buffEffect.setTargetPointer(new FixedTarget(source.getFirstTarget(), game));
+        game.addEffect(buffEffect, source);
+
+        // When that creature leaves the battlefield this turn, sacrifice Kjeldoran  Guard.
+        game.addDelayedTriggeredAbility(
+                new KjeldoranGuardDelayedTriggeredAbility(source.getFirstTarget()),
+                source);
+
+        return true;
+    }
+
+    private KjeldoranGuardEffect(KjeldoranGuardEffect effect) { super(effect); }
+
+    @Override
+    public KjeldoranGuardEffect copy() { return new KjeldoranGuardEffect(this); }
+}
+
+class KjeldoranGuardDelayedTriggeredAbility extends DelayedTriggeredAbility {
+
+    private final UUID creatureId;
+
+    KjeldoranGuardDelayedTriggeredAbility(UUID creatureId) {
+        super(new SacrificeSourceEffect(), Duration.EndOfTurn, true);
+        this.creatureId = creatureId;
+    }
+
+    KjeldoranGuardDelayedTriggeredAbility(KjeldoranGuardDelayedTriggeredAbility ability) {
+        super(ability);
+        this.creatureId = ability.creatureId;
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.ZONE_CHANGE;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        return event.getTargetId().equals(creatureId);
+    }
+
+    @Override
+    public KjeldoranGuardDelayedTriggeredAbility copy() {
+        return new KjeldoranGuardDelayedTriggeredAbility(this);
+    }
+
+    @Override
+    public String getRule() { return "that creature left the battlefield this turn"; }
```

Comment:
I think something like "When that creature left the battlefield, sacrifice {this}." would be a little nicer on the delayed trigger message.

It could be done by setting `setTriggerPhrase` in the constructor for instance.